pkg/vetter/conflictingvirtualservicehost/vet.go

```diff
@@ -392,6 +398,22 @@ func (v *VsHost) Vet() ([]*apiv1.Note, error) {
 	return notes, nil
 }
 
+// check same VS route conflict
+func sameVSconflict(rule1 routeRule, rule2 routeRule) (bool, error) {
+
+       if (rule1.priority > rule2.priority &&
+		((rule1.ruleType == prefix && (rule2.ruleType == prefix || rule2.ruleType == exact)) ||
+			(rule1.ruleType == exact && rule2.ruleType == exact))) {
+		return strings.HasPrefix(rule2.route, rule1.route), nil
+
+	} else if (rule2.priority > rule1.priority &&
+		((rule2.ruleType == prefix && (rule1.ruleType == prefix || rule1.ruleType == exact)) ||
+			   (rule2.ruleType == exact && rule1.ruleType == exact))) {
+		return strings.HasPrefix(rule1.route, rule2.route), nil
```

Comment:
This is a good move in the right direction, but I have some concerns about readability.

The many clauses in the if statement are quite hard to read and process. I think I understand where you're coming from (what if we add a new type?)

However, if that's something that you're concerned about, instead of returning `false, nil` when it doesn't match this clause, a better way to address that would be to do something like what's done in the `conflict` function (specifically [this line](https://github.com/aspenmesh/istio-vet/blob/0904472e61adf81954e50375c417a3085ca0853a/pkg/vetter/conflictingvirtualservicehost/vet.go#L365)), which explicitly returns an error when there's a case that we don't handle.

That way, we can make the code easier to read _and_ make sure we're resilient against new route types being added. Even though, I be surprised if different rule types were added, it is a good thing to keep in mind.